opendrift/models/radionuclides.py

```diff
@@ -0,0 +1,620 @@
+            default_density =  self.get_config('radionuclide:sediment:sediment_density') # default particle density (kg/m3)
+            f           =  self.get_config('radionuclide:sediment:effective_fraction')      # fraction of effective sorbents
+            phi         =  self.get_config('radionuclide:sediment:corr_factor')      # sediment correction factor 
+            poro        =  self.get_config('radionuclide:sediment:porosity')      # sediment porosity 
+            layer_thick =  self.get_config('radionuclide:sediment:layer_thick')      # thickness of seabed interaction layer (m) 
+            
+            self.transfer_rates[self.num_lmm,self.num_prev] = Dc * Kd * susp_mat
+            self.transfer_rates[self.num_prev,self.num_lmm] = Dc 
+            self.transfer_rates[self.num_lmm,self.num_srev] = \
+                Dc * Kd * sedmixdepth * default_density * (1.-poro) * f * phi / layer_thick
+            self.transfer_rates[self.num_srev,self.num_lmm] = Dc * phi 
+            self.transfer_rates[self.num_srev,self.num_ssrev] = slow_coeff
+            self.transfer_rates[self.num_prev,self.num_psrev] = slow_coeff
+            self.transfer_rates[self.num_ssrev,self.num_srev] = slow_coeff*.1
+            self.transfer_rates[self.num_psrev,self.num_prev] = slow_coeff*.1
+
+        elif transfer_setup=='dummy':
+        # Set of dummy values for testing/development
+        
+            self.num_lmm   = self.specie_name2num('LMM')
+            if self.get_config('radionuclide:species:Colloid'):
+                self.num_col = self.specie_name2num('Colloid')
+            self.num_prev  = self.specie_name2num('Particle reversible')
+            self.num_srev  = self.specie_name2num('Sediment reversible')
+            if self.get_config('radionuclide:slowly_fraction'):
+                self.num_psrev  = self.specie_name2num('Particle slowly reversible')
+                self.num_ssrev  = self.specie_name2num('Sediment slowly reversible')
+            
+
+            self.transfer_rates[self.num_lmm,self.num_prev] = 1.e-5 #*0.
+            self.transfer_rates[self.num_srev,self.num_lmm] = 5.e-6 
+            if self.get_config('radionuclide:slowly_fraction'):
+                self.transfer_rates[self.num_prev,self.num_psrev] = 2.e-6
+                self.transfer_rates[self.num_srev,self.num_ssrev] = 2.e-6
+                self.transfer_rates[self.num_psrev,self.num_prev] = 2.e-7
+                self.transfer_rates[self.num_ssrev,self.num_srev] = 2.e-7
+            
+        else:
+            logging.ERROR('No transfer setup available')
+
+        
+        # Set diagonal to 0. (not possible to transform to present specie)
+        np.fill_diagonal(self.transfer_rates,0.)
+        logging.info('nspecies: %s' % self.nspecies)
+        logging.info('Transfer rates: %s' % self.transfer_rates)
+
+
+
+
+
+
+        
+
+    def update_terminal_velocity(self, Tprofiles=None,
```

Comment:
This method seems to be copied directly from pelagicegg.py. It unfortunate to duplicate code, so ideally it should be either imported from pelagicegg.py, or both these modules should import it from another module/class. But this can also be done later.

Reply:
Indeed, it is copied. However it is slightly modified, using particle density from a preset element property, instead of computing it. I agree that it is not fortunate to have duplicates.